### PR TITLE
fix(log) : Add validation for when the error faced when the service is called did not result in an HTTP response 

### PR DESCRIPTION
**Pull request by camilamacedo86**
## Motivation
The scenario was checked by running the app locally. 

## What
Add validation for when the error faced when the service is called did not result in an HTTP response 

## Why
Avoid panic because the response is nil for example when debugging it locally. 

## Verification Steps
Check the CI

## Checklist:

- [X] Code has been tested locally by PR requester
- [ ] Changes have been successfully verified by another team member 

## Progress

- [x] Finished task
- [ ] TODO

### CODE REVIEW

File: pkg/service/client.go [hunk 1]
```diff
@@ -28,7 +28,10 @@ var DeleteAppFromServiceByRestAPI = func(serviceAPI string, id string, reqLogger
 	response, err := client.Do(req)
 
 	if err != nil || 204 != response.StatusCode {
```
Comment (davidffrench) [inline pkg/service/client.go]: @camilamacedo86 If response is nil, this line of `response.StatusCode` will just throw a runtime error.

Comment (camilamacedo86) [reply pkg/service/client.go]: Yes. I will fix it properly and ping you.

**Comment by camilamacedo86** [comment]
@laurafitzgerald @austincunningham could you please check this one for we move forward here.

File: pkg/service/client.go [hunk 2]
```diff
@@ -99,7 +105,10 @@ func GetAppFromServiceByRestApi(serviceAPI string, appId string, reqLogger logr.
 	client := &http.Client{}
 	response, err := client.Do(req)
 	if err != nil {
-		reqLogger.Error(err, "Unable to execute GET request", "HTTPMethod", http.MethodGet, "url", url, "response.StatusCode", response.StatusCode)
+		if response != nil {
+			reqLogger.Error(err, "Unable to execute GET request", "HTTPMethod", http.MethodGet, "url", url, "response.StatusCode", response.StatusCode)
+		}
+		reqLogger.Error(err, "Unable to execute GET request", "HTTPMethod", http.MethodGet, "url", url)
```
Comment (austincunningham) [inline pkg/service/client.go]: not sure you need to do this here as nil is being returned

Comment (camilamacedo86) [reply pkg/service/client.go]: When the issue occurs the response is nil. In this case, the response can be nil which will cause panic.

**Pull request merged**